jellyfin_mpv_shim/rich_presence.py

```diff
@@ -28,16 +40,32 @@ def send_presence(
         state=subtitle,
         details=title,
         instance=False,
-        large_image="jellyfin2",
+        large_image=upload_image(artwork_url),
         start=start,
         end=end,
-        large_text="Jellyfin",
+        large_text=title,
         small_image=small_image,
+        small_text=small_text,
         party_id=str(hash(syncplay_group)),
-        party_size=[1, 100],
         join=syncplay_group,
     )
 
 
 def clear_presence():
     RPC.clear_activity()
+
+
+def upload_image(link):
+    global image_url
+    global bashupload_url
+    if image_url == link:
+        return bashupload_url
+    image_url = link
+    r = requests.get(link)
+    files = {"file": ("image.jpg", r.content)}
+    post = requests.post("https://bashupload.com/", files=files)
+
+    regex = r"https(.*)"
+    result = re.search(regex, post.text)
+    bashupload_url = result.group(0)
+    return bashupload_url
```

Comment:
Several problems with this:
 - This has privacy implications for the end-user.
 - The `get_timeline_options` executes every 5 seconds and also in response to user events. This would be abusive to this service, which probably does not want software automatically hammering it with uploads.
 - What happens if the request fails? Does it crash the timeline handler or hang it indefinitely? We definitely do not want to cause timeline updates to fail since it can leave the player stuck in an outdated state.
 - If a Jellyfin server is public, you don't even need an authentication key to grab the image, so it could just be sent to discord as-is. The application already knows if a server URL is public (there is a function in utils that can check this).

Reply:
- What would you suggest? A toggle for uploading the image?
- It did not crash it, but now if the upload fails it just returns the default Jellyfin icon.
- Done, only uploads when the server is private.